src/Sulu/Bundle/CoreBundle/SuluCoreBundle.php

```diff
@@ -47,5 +51,11 @@ public function build(ContainerBuilder $container)
                 'alias'
             )
         );
+
+        $container->addCompilerPass(
+            ServiceCustomizerCompilerPass::customize('fos_rest.serializer.exception_normalizer.jms')
```

Comment:
What are we really saving with this CompilerPass? I think just using the Symfony stuff here is not a lot more code (I think only the check if the service is existing can be omitted), and it is more complicated to use (and we have to maintain the documentation, tests and the code itself...)

Reply:
these compiler-passes helps to overwrite and extend existing services of another bundle

Reply:
I know what it does, but I don't see the advantage of defining a separate class for that. The resulting code almost isn't smaller, so I would tend to do it the old way.

Reply:
what is the old way? over xml definition?

Reply:
I've had a closer look, and realized that we don't need that at all 😉 
The [`UPGRADE.md`](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/d4bd99f7f0bcf8d8e765d70186bdfa4f42ea9ea2/UPGRADING-2.0.md) doesn't say this should be done this way. What they seem to suggest is that another handler for the serializer is defined, which could probably be done without that additional CompilerPass, with a simple service definition.

Reply:
Or maybe even use a [service decorator](https://symfony.com/doc/current/service_container/service_decoration.html)?

Reply:
the problem is that the normalizer won't work with jms/serializer and decorator also don't work because we override a protected functions.